backend/Microservices/AuthMicroservice/Auth.UnitTests/UsersTests/UsersTests.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Auth.Application.Settings;
+
+namespace Auth.UnitTests.UsersTests
+{
+    public class UsersTests : UserServiceTestBase
+    {
+        [Fact]
+        public async Task GetAllAsync_WhenCachedUsersExist_ShouldReturnCachedUsers()
+        {
+            // Arrange
+            var paginationSettings = Create<PaginationSettings>();
+            var cachedUsers = CreateMany<UserModel>(2).ToList();
+
+            _cacheRepositoryMock.Setup(cr => cr.GetCachedLargeDataAsync<UserModel>(paginationSettings))
+                .ReturnsAsync(cachedUsers);
+
+            // Act
+            var result = await _usersService.GetAllAsync(paginationSettings, CancellationToken.None);
+
+            // Assert
+            result.Should().BeEquivalentTo(cachedUsers);
+            _unitOfWorkMock.Verify(uw => uw.Users.GetAllAsync(paginationSettings, CancellationToken.None), Times.Never);
+        }
+
+        [Fact]
+        public async Task GetAllAsync_WhenNoCachedUsersExists_ShouldRetrieveUsersFromDatabase()
+        {
+            // Arrange
+            var paginationSettings = Create<PaginationSettings>();
+            paginationSettings.PageNumber = 1;
```

Comment:
it's better to set params directly via autofixture. It's not obligatory to use template create
![image](https://github.com/user-attachments/assets/2db9ee33-7cfe-4333-993d-c20b294c6324)